examples_tests/35.GeometryCreator/main.cpp

```diff
@@ -97,17 +99,49 @@ int main()
 	auto queues = std::move(initOutput.queues);
 	auto swapchain = std::move(initOutput.swapchain);
 	auto renderpass = std::move(initOutput.renderpass);
-	auto fbo = std::move(initOutput.fbo[0]);
+	auto fbos = std::move(initOutput.fbo);
 	auto commandPool = std::move(initOutput.commandPool);
 	auto assetManager = std::move(initOutput.assetManager);
 	auto logger = std::move(initOutput.logger);
 	auto inputSystem = std::move(initOutput.inputSystem);
 	auto windowCallback = std::move(initOutput.windowCb);
-	auto cpu2gpuParams = std::move(initOutput.cpu2gpuParams);
+
 	nbl::video::IGPUObjectFromAssetConverter cpu2gpu;
+	nbl::video::IGPUObjectFromAssetConverter::SParams cpu2gpuParams;
+
+	nbl::core::smart_refctd_ptr<nbl::video::IGPUFence> gpuTransferFence;
+	nbl::core::smart_refctd_ptr<nbl::video::IGPUSemaphore> gpuTransferSemaphore;
+
+	nbl::core::smart_refctd_ptr<nbl::video::IGPUFence> gpuComputeFence;
+	nbl::core::smart_refctd_ptr<nbl::video::IGPUSemaphore> gpuComputeSemaphore;
+
+	auto updateCpu2GpuSignalizatorsWithPureObjects = [&]() -> void //! reset the state by creating new gpu objects after cpu2gpu conversion
+	{
+		gpuTransferFence = logicalDevice->createFence(static_cast<video::IGPUFence::E_CREATE_FLAGS>(0));
+		gpuTransferSemaphore = logicalDevice->createSemaphore();
+
+		gpuComputeFence = logicalDevice->createFence(static_cast<video::IGPUFence::E_CREATE_FLAGS>(0));
+		gpuComputeSemaphore = logicalDevice->createSemaphore();
+	};
```

Comment:
same question as before

Reply:
what do you mean?

Reply:
this sample works

Reply:
ok, GitHub failed me again

didn't we solve the fence thing last week?

Also you can reuse semaphores just fine, you just need each signal and wait to come in pairs and always happen (signal must be consumed).

Reply:
like we shouldn't need to create fresh fences anymore, just use the same one for all transfers and await at the end after the last and manual submit.

Reply:
> ok, GitHub failed me again
> 
> didn't we solve the fence thing last week?
> 
> Also you can reuse semaphores just fine, you just need each signal and wait to come in pairs and always happen (signal must be consumed).

yeah we did it, actually I should use those semaphoes, fences because they do nothing at all (I should use their signalized state for some wait)